### ISSUE TITLE
Device appears to freeze randomly, but remains responsive otherwise.

### ISSUE THREAD

**Issue by vonmatterlorenzohorn** [ExplosiveLobster/bug_tracker#25]
**Device:*F5122*
**Kernel version:*4.9*
**Android version:*11*

**Describe the bug:**
With the latest build (without other GSI installed), device randomly freezes on a screen. Interactions do happen(I can lock the screen and swipe up to unlock-it makes locking and unlocking sounds) but screen remains frozen.

**To Reproduce:**
Error is Random. Can happen when only 1 app is open, can happen when 5 background apps are open.
**System/Kernel logs:**
Attach system/kernel logs (logcat/dmesg) here.
[logcat.txt](https://github.com/ExplosiveLobster/bug_tracker/files/5945847/logcat.txt)

**Additional context:**
Was using Retro Music Player when log was recorded, bluetooth and WiFi connected, battery saver on. Using Pixel Launcher and Pixelize addons with NikGapps.

**Comment by mase76** [comment]
I can confirm this. Sometimes I get an additional screen flickering. That seems to be the biggest issue with this rom, which makes it impossible for a daily use. I think, this is kernel related. I also had such freezes with the last Omnirom builds, when they switched to a 4.x kernel.

**Comment by TechnoHT** [comment]
I think this is OEM binaries issues

**Comment by TechnoHT** [comment]
04-06 18:00:23.151   598   598 E RenderEngine: failed to wait for EGL native fence sync: 0x300c
04-06 18:00:23.151   598   598 E RenderEngine: failed to wait on fence fd
04-06 18:00:23.151   598   598 W Adreno-GSL: <gsl_ldd_control:548>: ioctl fd 12 code 0xc040094a (IOCTL_KGSL_GPU_COMMAND) failed: errno 35 Resource deadlock would occur
04-06 18:00:23.151   598   598 W Adreno-GSL: <log_gpu_snapshot:457>: panel.gpuSnapshotPath is not set.not generating user snapshot

**Comment by 57op** [comment]
@TechnoHT thanks for posting the logcat here, too.

today I managed to get kernel log as well:
> [ 2467.697220] kgsl kgsl-3d0: surfaceflinger[1408]: gpu timeout ctx 2 ts 24619
> [ 2467.697285] kgsl kgsl-3d0: surfaceflinger[1408]: gpu fault ctx 2 ctx_type ANY ts 24619 status 800001C1 rb 18dc/18dc ib1 00000000F8000400/0000 ib2 00000000BEAFD348/0000
> [ 2467.697292] kgsl kgsl-3d0: surfaceflinger[1408]: gpu fault rb 0 rb sw r/w 18dc/1906
> [ 2467.715675] kgsl kgsl-3d0: |adreno_hang_int_callback| MISC: GPU hang detected
> [ 2467.805974] kgsl kgsl-3d0: |kgsl_device_snapshot| snapshot created at pa 0x00000000c9600000 size 520196
> [ 2467.806259] kgsl kgsl-3d0: surfaceflinger[1408]: gpu failed ctx 2 ts 24619
> [ 2467.808583] kgsl kgsl-3d0: |kgsl_snapshot_save_frozen_objs| snapshot: Active IB1:00000000f8000400 not dumped

it might be a kernel problem, but i'm no expert.
i'll try compiling the kernel reverting this commit https://github.com/ExplosiveLobster/kernel_sony_msm-4.9/commit/3c6d7e5db9b371cd886aa44a5a34a15908fb869f and see what happens.

update:
kernel has been successfully compiled, i will be trying it in the following days and report if any freeze occurs.
if anyone is interested in testing, i can share a zip file that patches boot.img

**Comment by TechnoHT** [comment]
> it might be a kernel problem, but i'm no expert.
i'll try compiling the kernel reverting this commit ExplosiveLobster/kernel_sony_msm-4.9@3c6d7e5 and see what happens.

I tried a few solutions focusing on Adreno driver, most of them failed to work :b, but haven't tried this one
looking for the result

**Comment by vonmatterlorenzohorn** [comment]
Apparently this is an issue due to magisk according to a ROM dev I told this to. Can anyone else who has this issue confirm that they have magisk installed?

**Comment by 57op** [comment]
> I tried a few solutions focusing on Adreno driver, most of them failed to work :b, but haven't tried this one
> looking for the result

well, it was a failure.. the bug has just occured

> Apparently this is an issue due to magisk according to a ROM dev I told this to. Can anyone else who has this issue confirm that they have magisk installed?

yes, i can confirm. i've always used the rom with magisk

**Comment by TechnoHT** [comment]
> Apparently this is an issue due to magisk according to a ROM dev I told this to. Can anyone else who has this issue confirm that they have magisk installed?

Yep, I always has Magisk, maybe I should try to looking outside of Adreno driver

**Comment by TechnoHT** [comment]
@vonmatterlorenzohorn can you give us a bit more information? following the only clue that it has to be something related to Magisk makes me feel a bit confusing.

**Comment by vonmatterlorenzohorn** [comment]
> @vonmatterlorenzohorn can you give us a bit more information? following the only clue that it has to be something related to Magisk makes me feel a bit confusing.

Sorry I can't. According to the guy I talked to, this happens from android 9 and up. I've experienced this same issue with my current phone, the Lenovo phone I maintain lineage for and the Xperia X. For me it has always occured on android 11, but some people say it also happens in android 10 and 9.

**Comment by bartcubbins** [comment]
as far as I remember, the problem is triggered by Adreno blobs

**Comment by vonmatterlorenzohorn** [comment]
> as far as I remember, the problem is triggered by Adreno blobs

Interesting..

**Comment by eMPee584** [comment]
💭 [android is the _worst_ bitchy clusterfuck **ever**.. what a debugging nightmare.. big kudos for your persistence guys]

**Comment by bartcubbins** [comment]
it's not even about Android, but the fact that a lot of things are proprietary which are very difficult or almost impossible to debug. I will try to help somehow as soon as I find the time and mood for this;)

**Comment by TechnoHT** [comment]
Pretty odd thing is that when I get the kernel log, It almost identical as what @57op got,
but :

`[ 2467.715675] kgsl kgsl-3d0: |adreno_hang_int_callback| MISC: GPU hang detected`

sometimes it wasn't appear, and logcat still give me 0x300c
